R/parse.R

```diff
@@ -135,3 +138,47 @@ parse_check_consistent_dimensions_lhs <- function(eq, dat, call) {
     }
   }
 }
+
+find_ranks_in_expr <- function(expr, reduction = FALSE) {
```

Comment:
I don't think this strategy is the right one - collecting up usages and then throwing some combination. It will likely be simpler to just throw on the first error - see main review comment